src/Confluent.Kafka/Serialization/Extensions/Message.cs

```diff
@@ -38,14 +40,37 @@ public static class MessageExtensions
         ///     A typed message instance corresponding to <paramref name="message" />.
         /// </returns>
         public static Message<TKey, TValue> Deserialize<TKey, TValue>(this Message message, IDeserializer<TKey> keyDeserializer, IDeserializer<TValue> valueDeserializer)
-            => new Message<TKey, TValue> (
+        {
+            TKey key;
+            TValue val;
+
+            try
+            {
+                key = keyDeserializer.Deserialize(message.Key);
```

Comment:
Yep but keeping this PR to the consumer which is most important as people may hit this by running the examples.